### ISSUE TITLE
[Library Stabilization] Should there exist ** with a bigint as the exponent but the base is an integral?

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#17689]
The BigInteger module provides `**` (exponentiation) of a bigint base to a bigint exponent.  It seems like if that's supported, it should be pretty straight-forward to support an int or uint base to a bigint exponent.

In other bigint methods where there wasn't originally an int/uint equivalent, we wrap the int/uint in a bigint and then call the normal bigint method.  That should be tractable for this case as well, so far as I know.

**Comment by lydia-duncan** [comment]
Michael sounded in favor of this

**Comment by bradcray** [comment]
It seems like we should, though mostly it makes me wish we had `int->bigint` and `uint->bigint` coercions so that these cases would just "fall out" rather than requiring explicit overloads.  Was that possibility discussed?

Would there be any reason to not also support `integral**bigint`?

This doesn't technically seem like a library stabilization issue to me, in that it seems like a non-breaking change?  Is it in a way that I'm missing?

**Comment by lydia-duncan** [comment]
> It seems like we should, though mostly it makes me wish we had `int->bigint` and `uint->bigint` coercions so that these cases would just "fall out" rather than requiring explicit overloads. Was that possibility discussed?

I don't think we discussed it.  There are cast functions, but not coercions, I believe.  Shall I open an issue for that?

> Would there be any reason to not also support `integral**bigint`?

I believe that's the direction I'm proposing (but I didn't put what the call would look like in code, sorry about that)
 
> This doesn't technically seem like a library stabilization issue to me, in that it seems like a non-breaking change? Is it in a way that I'm missing?

Oops.  Good point.  I saw this yesterday and tried to adjust later ones so I wouldn't need to edit them as well

**Comment by bradcray** [comment]
> I believe that's the direction I'm proposing (but I didn't put what the call would look like in code, sorry about that)

Oh, you're right.  What about the opposite then:  `bigint**integral`?

> Shall I open an issue for that?

I think that could be interesting.  It's probably not a breaking change, but it seems like it could remove some redundancy in the code base for cases where we're just taking the integral, converting to bigint, and calling the bigint version?  And it'd be an issue in which coercions seem very reasonable to carry along the conversation we started at MS15.

**Comment by lydia-duncan** [comment]
> > I believe that's the direction I'm proposing (but I didn't put what the call would look like in code, sorry about that)
> 
> Oh, you're right. What about the opposite then: `bigint**integral`?

It doesn't seem like there'd be a problem with that.  It'd be pretty easy to implement, with just turning the exponent into a `bigint` before calling the normal `bigint` version
